### ISSUE TITLE
difference between tf hub and native tf code for efficientnet

### ISSUE THREAD

**Issue by jdvylder** [tensorflow/hub#697]
I originally was using the efficientnet B0 module (https://tfhub.dev/google/efficientnet/b0/feature-vector/1) in TF1.15.2 (nvcr.io/nvidia/tensorflow:20.03-tf1-py3 ). At some point I wanted to do an experiment where I needed access to certain hidden layers, and decided to swap the tf.hub module with the native tf code. Based on the documentation, I found that training of the hub module is supposed to be done using the official efficientnet repo (https://github.com/tensorflow/tpu/tree/master/models/official/efficientnet) with auto-augment preprocessing (so I assume with the following checkpoint: https://storage.googleapis.com/cloud-tpu-checkpoints/efficientnet/ckptsaug/efficientnet-b0.tar.gz) However when I try to train this new implementation I receive NaN errors during training, which I never received in the tf hub implementation. Btw his observation is not based on a single run. 
I know I can solve this by lowering the learning rate, but I don't understand why I would need to adjust my hyperparameters as I expect both implementations to execute the same network. Is this a bug or is their a logical explanation to this different behavior? 

The code snippet for the native TF code:
```
model_builder = model_builder_factory.get_model_builder("efficientnet-b0")
net, _ = model_builder.build_model(
    input_tensor,
    model_name="efficientnet-b0",
    training=mode == tf.estimator.ModeKeys.TRAIN,
    model_dir="/scratch/checkpoints/efficientnet-b0-aa/efficientnet-b0")
net = tf.expand_dims(net, 1)
net = tf.expand_dims(net, 1)
with tf.variable_scope('classification_head'):
    drop = tf.layers.dropout(
        net, rate=self.configuration.get("dropout"),
        seed=self.configuration.get("random_seed"),
        training=mode == tf.estimator.ModeKeys.TRAIN
    )
logits_classification = tf.layers.conv2d(drop,9, 1, activation=None, name='fc_logits')
logits_classification = tf.squeeze(logits_classification, [1, 2], name='SpatialSqueeze')
…
```

Where warm variables in the estimator are defined using ".*efficientnet-b.*" and all trainable variable either contain “classification_head” or “efficientnet”

De code snippet voor de tf hub versie:
```
if mode == tf.estimator.ModeKeys.TRAIN:
    tags_ = {"train"}
else:
    tags_ = {}
module = hub.Module("https://tfhub.dev/google/efficientnet/b0/feature-vector/1", trainable=True, tags=tags_)
net = module(input_tensor)
with tf.variable_scope('classification_head'):
    drop = tf.layers.dropout(
        net, rate=self.configuration.get("dropout"),
        seed=self.configuration.get("random_seed"),
        training=mode == tf.estimator.ModeKeys.TRAIN
     )
logits_classification = tf.layers.conv2d(drop,9, 1, activation=None, name='fc_logits')
logits_classification = tf.squeeze(logits_classification, [1, 2], name='SpatialSqueeze')
…
```
With warm variables in the estimator are defined using ".*" and all trainable variables either contain “classification_head” or “efficientnet”

Thanks for looking into this.

Kr

Jonas

**Comment by MorganR** [comment]
Hello @jdvylder, thank you for reaching out. Both the TF Hub and native code should provide similar results. One thing to note is that the TF Hub version expects inputs in the range from [0,1], following Hub's [common image input conventions](https://www.tensorflow.org/hub/common_saved_model_apis/images#image_input). However, the native code expects image values in the range [0, 255].

Can you confirm if that resolves your issue?

**Comment by MorganR** [comment]
Closing due to lack of response. Please reopen if further discussion is required.
